src/layout/Group/GroupContainer.tsx

```diff
@@ -166,6 +164,13 @@ export function GroupContainer({ id }: IGroupProps): JSX.Element | null {
     );
   }
 
+  const displayBtn = () =>
+    edit?.mode !== 'showAll' &&
+    edit?.addButton !== false &&
+    'maxCount' in node.item &&
+    repeatingGroupIndex + 1 < (node.item.maxCount === undefined ? -99 : node.item.maxCount) &&
+    (editIndex < 0 || edit?.alwaysShowAddButton === true);
```

Comment:
This doesn't have to be a function, you can just assign the value to the `displayBtn` constant instead.

The check for `edit?.mode !== 'showAll'` is just because we render `<AddButton>` in two different places (see the other code that renders AddButton below here). It's a bit misleading to say that we don't display the add button when the mode is `showAll`, because we absolutely do. Maybe you could re-use the `displayBtn` boolean there as well, so that we get this functionality when using `edit.mode === 'showAll'` (as an app developer I would expect the functionality to work the same regardless of edit mode).